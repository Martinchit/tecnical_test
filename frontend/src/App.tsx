import React from 'react';
import { BrowserRouter as Router, Redirect, Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import HomePage from './pages/Home/HomePage';
import StockListPage from './pages/StockList/StockListPage';
import OrdersBasketPage from './pages/OrdersBasket/OrdersBasketPage';
import { NavBar } from './core/components/organisms/NavBar';
import { FlexBox } from './core/components/atoms/FlexBox';
import { RootStateType, ActionsType, actions } from './redux/store';

interface Props {
  loggedIn: boolean;
  logOutAction: () => void;
}

export const App: React.FC<Props> = ({ loggedIn, logOutAction }) => {
  return (
    <Router>
      <NavBar loggedIn={loggedIn} logOutAction={logOutAction} />
      <FlexBox margin="2% 10%" width="80%">
        {loggedIn ? (
          <Switch>
            <Route exact path="/stock_list" component={StockListPage} />
            <Route exact path="/orders_basket" component={OrdersBasketPage} />
            <Redirect from="*" to="/stock_list" />
          </Switch>
        ) : (
          <Switch>
            <Route exact path="/" component={HomePage} />
            <Redirect from="*" to="/" />
          </Switch>
        )}
      </FlexBox>
    </Router>
  );
};

const mapStateToProps = (state: RootStateType) => ({
  loggedIn: state.auth.loggedIn,
});

const mapDispatchToProps = (dispatch: Dispatch<ActionsType>) => ({
  logOutAction: () => dispatch(actions.logOut()),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
