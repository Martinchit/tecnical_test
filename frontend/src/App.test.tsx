import React from 'react';
import { shallow } from 'enzyme';
import { Route, Redirect } from 'react-router-dom';

import { App } from './App';

describe('App', () => {
  const logOutAction = jest.fn();
  it('renders correctly', () => {
    const loggedIn = false;
    const wrapper = shallow(<App loggedIn={loggedIn} logOutAction={logOutAction} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders routes before logged in', () => {
    const loggedIn = false;
    const wrapper = shallow(<App loggedIn={loggedIn} logOutAction={logOutAction} />);
    expect(wrapper.find(Route).length).toBe(1);
    expect(wrapper.find(Redirect).props().to).toBe('/');
  });

  it('renders routes after logged in', () => {
    const loggedIn = true;
    const wrapper = shallow(<App loggedIn={loggedIn} logOutAction={logOutAction} />);
    expect(wrapper.find(Route).length).toBe(2);
    expect(wrapper.find(Redirect).props().to).toBe('/stock_list');
  });
});
