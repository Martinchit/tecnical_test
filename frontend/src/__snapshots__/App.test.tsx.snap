// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<BrowserRouter>
  <Component
    logOutAction={[MockFunction]}
    loggedIn={false}
  />
  <styled.div
    margin="2% 10%"
    width="80%"
  >
    <Switch>
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Home)",
            "type": [Function],
          }
        }
        exact={true}
        path="/"
      />
      <Redirect
        from="*"
        to="/"
      />
    </Switch>
  </styled.div>
</BrowserRouter>
`;
